### ISSUE TITLE
Bug bounty program

### ISSUE THREAD

**Issue by Changaco** [liberapay/liberapay.org#59]
[Liberapay's HackerOne program](https://hackerone.com/liberapay) was launched in 2018 (<https://github.com/liberapay/liberapay.com/issues/549>), and I don't think the bounty amounts have been changed since. Liberapay has significantly more income now than it did then, so we could increase the bounty amounts.

@karelorigin has proposed a simple doubling of the current amounts. Are there any other proposals? @EdOverflow?

(The best time for a significant update of the HackerOne program would probably be after Liberapay migrates away from AWS and SQL, as there will be a greater need for reviews of the new infrastructure's security. However, that would mean waiting quite a while, as the work to make that migration possible is nowhere near complete.)

**Comment by EdOverflow** [comment]
I agree with @karelorigin that doubling the current ranges is a step in the right direction. This may encourage some of the more reputable hackers on the platform to review Liberapay.

**Comment by karelorigin** [comment]
@Changaco since you have a better understanding of Liberapay's budget, I'd suggest taking it as a minimum. I think a more is better approach is generally preferred when it comes to bounties.

**Comment by Changaco** [comment]
> I don't think the bounty amounts have been changed since.

Actually, [I changed them last year](https://hackerone.com/liberapay/bounty_table_versions?change=2022-10-09T09%3A40%3A22.788Z), so almost all of the maximum amounts have already been doubled.

**Comment by Changaco** [comment]
I'm widening the scope of this issue to the bug bounty program in general, including the possibility of moving from HackerOne to a different platform.

The main problem I currently have with HackerOne is that the person who reports a vulnerability can block its disclosure on HackerOne, which has led to multiple reports staying hidden. Of course we could publish report summaries elsewhere, for example as GitHub security advisories or on the Liberapay website, but we shouldn't have to. HackerOne supports publishing summaries instead of the full report, yet even that can be blocked by the reporter, which makes no sense to me.

Another thing that's been bothering me for years is that researchers are expected to run their own scans of a website when looking for vulnerabilities. This is bad for the scanned website because it receives a lot more junk requests than necessary, and it's bad for the researchers because it wastes their time and resources.

**Comment by Changaco** [comment]
I've disabled submissions on HackerOne. Previously the program was marked as accepting only reports of critical vulnerabilities, but that wasn't stopping non-critical reports from coming in.

I'm not planning to reopen until at least the end of August, and I would like to fully delegate the processing of reports to someone else when submissions are reopened.
